src/util/routes.js

```diff
@@ -20,8 +20,11 @@ const toPathByRouteName = (nameToFind, routes) => {
 /**
  * Shorthand for single path call. (```pathByRouteName('ListingPage', routes, { id: uuidX });```)
  */
-export const pathByRouteName = (nameToFind, routes, params = {}) =>
-  toPathByRouteName(nameToFind, routes)(params);
+export const pathByRouteName = (nameToFind, routes, params = {}) => {
+  const hasEmptySlug = params && Object.keys(params).find(k => k === 'slug') && params.slug === '';
```

Comment:
No need to iterate the keys:

```suggestion
  const hasEmptySlug = params && params.hasOwnProperty('slug') && params.slug === '';
```

Reply:
Ah, true. I should stay in my bed.